packages/fx-core/src/plugins/resource/function/constants.ts

```diff
@@ -165,8 +165,8 @@ export class DefaultProvisionConfigs {
 }
 
 export class AzureInfo {
-  public static readonly storageAccountNameLenMax: number = 24;
-  public static readonly functionAppNameLenMax: number = 32;
+  public static readonly resourceNameLenMax: number = 24;
+  public static readonly suffixLenMax: number = 12;
```

Comment:
Why 12?

Reply:
no reason, just pick a number that is half of resourceNameLenMax. I think the value only has two requirements:
1. no more than half of the full length, because suffix string is not human-readable.
2. no less than our current suffix length (6).